.github/workflows/ci.yml

```
@@ -69,27 +92,27 @@ jobs:
 
       - name: Configure CMake
         run: |
-          cmake -S . -B ./build -DCMAKE_BUILD_TYPE:STRING=${{env.BUILD_TYPE}}
+          cmake -S . -B ./build -DCMAKE_BUILD_TYPE:STRING=${{matrix.build_type}} -DENABLE_DEVELOPER_MODE:BOOL=${{matrix.developer_mode}}
```

Let's use Ninja for everything. Making Ninja is always the best thing to do as it avoids any implicit dependency on system libraries and headers (unlike MSVC generator). setup-cpp already runs vcvarsall on Windows, so using Ninja should work.

```suggestion
          cmake -S . -B ./build -G "Ninja Multi-Config" -DCMAKE_BUILD_TYPE:STRING=${{matrix.build_type}} -DENABLE_DEVELOPER_MODE:BOOL=${{matrix.developer_mode}}
```

---

I want to make sure that MSVC's generator is not completely removed, so I'll probably update this again after merging your suggested change.